Lib/enum.py

```diff
@@ -926,35 +1085,72 @@ def _generate_next_value_(name, start, count, last_values):
 
     @classmethod
     def _missing_(cls, value):
-        """
-        Returns member (possibly creating it) if one can be found for value.
-        """
-        original_value = value
-        if value < 0:
-            value = ~value
-        possible_member = cls._create_pseudo_member_(value)
-        if original_value < 0:
-            possible_member = ~possible_member
-        return possible_member
-
-    @classmethod
-    def _create_pseudo_member_(cls, value):
         """
         Create a composite member iff value contains only members.
         """
-        pseudo_member = cls._value2member_map_.get(value, None)
-        if pseudo_member is None:
-            # verify all bits are accounted for
-            _, extra_flags = _decompose(cls, value)
-            if extra_flags:
-                raise ValueError("%r is not a valid %s" % (value, cls.__qualname__))
+        if not isinstance(value, int):
+            raise ValueError("%r is not a valid %s" % (value, cls.__qualname__))
+        # check boundaries
+        # - value must be in range (e.g. -16 <-> +15, i.e. ~15 <-> 15)
+        # - value must not include any skipped flags (e.g. if bit 2 is not defined, then 0d10 is invalid)
+        neg_value = None
+        if (
+                not ~cls._all_bits_ <= value <= cls._all_bits_
+                or value & (cls._all_bits_ ^ cls._flag_mask_)
+            ):
+            if cls._boundary_ is STRICT:
+                invalid_as_bits = _bits(value)
+                length = max(len(invalid_as_bits), _bit_len(cls._flag_mask_))
+                valid_as_bits = ('0' * length + _bits(cls._flag_mask_))[-length:]
+                invalid_as_bits = ('01'[value<0] * length + invalid_as_bits)[-length:]
+                raise ValueError(
+                        "%s: invalid value: %r\n    given %s\n  allowed %s" % (
+                            cls.__name__,
+                            value,
+                            invalid_as_bits,
+                            valid_as_bits,
+                            ))
+            elif cls._boundary_ is CONFORM:
+                value = value & cls._flag_mask_
+            elif cls._boundary_ is EJECT:
+                return value
+            elif cls._boundary_ is KEEP:
+                if value < 0:
+                    value = max(cls._all_bits_+1, 2**(value.bit_length())) + value
+            else:
+                raise ValueError('unknown flag boundary: %r' % (cls._boundary_, ))
+        if value < 0:
+            neg_value = value
+            value = cls._all_bits_ + 1 + value
+        # get members
+        members, unknown = _decompose(cls, value)
```

Comment:
why is decompose still needed?

Reply:
It does the work of figuring out which bits are valid flags.  I did switch it to walking the bits of `value`, and not worrying about negative numbers (since it is not called with any) and it is now about 4 times faster.

Reply:
- Done.
- Moved to class as `_iter_member_by_value_`, aka `_iter_member_`.
- `yield from self._iter_member_()`

Note that if definition order is not the same as lsb order, `_iter_member_by_def_` is used instead of `_iter_member_by_value_`.